app/services/people/retrieve_court_cases.rb

```diff
@@ -2,17 +2,33 @@
 
 module People
   class RetrieveCourtCases
-    def self.call(person, filter_params)
-      nomis_court_cases_response = NomisClient::CourtCases.get(person.latest_nomis_booking_id, filter_params)
+    class << self
+      def call(person, filter_params)
+        check_presence_of_latest_nomis_booking_id(person)
 
-      court_cases = JSON.parse(nomis_court_cases_response).map do |court_case|
-        CourtCase.new.build_from_nomis(court_case)
+        nomis_court_cases_response = NomisClient::CourtCases.get(person.latest_nomis_booking_id, filter_params)
+
+        court_cases = JSON.parse(nomis_court_cases_response).map do |court_case|
+          CourtCase.new.build_from_nomis(court_case)
+        end
+
+        OpenStruct.new(success?: true, court_cases: court_cases, errors: nil)
+      rescue OAuth2::Error => e
+        nomis_error = NomisClient::ApiError.new(status: e.response.status, error_body: e.response.body)
+        OpenStruct.new(success?: false, court_cases: [], error: nomis_error)
       end
 
-      OpenStruct.new(success?: true, court_cases: court_cases, errors: nil)
-    rescue OAuth2::Error => e
-      nomis_error = NomisClient::ApiError.new(status: e.response.status, error_body: e.response.body)
-      OpenStruct.new(success?: false, court_cases: [], error: nomis_error)
+    private
+
+      def check_presence_of_latest_nomis_booking_id(person)
```

Comment:
I have a strong suspicion that more than just this service uses latest_nomis_booking_id? I'm wondering whether it makes sense resolving them all here?